### PR TITLE
Sphinx Build Github Action on Release

### PR DESCRIPTION
**Pull request by devmrfitz**
# Description
Added github action to build docs on release

## Related issues

#804 

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue).
- [X] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).

# Checklist

- [X] I have read and understood the rules about [how to Contribute](https://intelowl.readthedocs.io/en/latest/Contribute.html) to this project
- [X] The pull request is for the branch `develop`
- [ ] A new analyzer or connector was added, in which case:
    - [ ] [Usage](https://github.com/intelowlproject/IntelOwl/blob/master/docs/source/Usage.md) file was updated.
    - [ ] [Advanced-Usage](./Advanced-Usage.md) was updated (in case the analyzer/connector provides additional optional configuration).
    - [ ] Secrets were added in [env_file_app_template](https://github.com/intelowlproject/IntelOwl/blob/master/docker/env_file_app_template), [env_file_app_ci](https://github.com/certego/IntelOwl/blob/master/docker/env_file_app_ci) and in the [Installation](./Installation.md) docs, if necessary.
    - [ ] If the analyzer/connector requires mocked testing, `_monkeypatch()` was used in it's class to apply the necessary decorators.
    - [ ] If a File analyzer was added, it's name was explicitly defined in [test_file_scripts.py](https://github.com/intelowlproject/IntelOwl/blob/master/tests/analyzers_manager/test_file_scripts.py) (not required for Observable Analyzers).
- [ ] If external libraries/packages with restrictive licenses were used, they were added in the [Legal Notice](https://github.com/certego/IntelOwl/blob/master/.github/legal_notice.md) section.
- [X] The tests gave 0 errors.
- [X] Linters (`Black`, `Flake`, `Isort`) gave 0 errors. If you have correctly installed [pre-commit](https://intelowl.readthedocs.io/en/latest/Contribute.html#how-to-start-setup-project-and-development-instance), it does these checks and adjustments on your behalf.
- [X] The commits were squashed into a single one (optional, they will be squashed anyway by the maintainer)
  
### Important Rules
- If your changes decrease the overall tests coverage (you will know after the Codecov CI job is done), you should add the required tests to fix the problem
- Everytime you make changes to the PR and you think the work is done, you should explicitly ask for a review

### CODE REVIEW

File: .github/workflows/sphinx_build_on_release.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+      - uses: actions/checkout@v2
+      - name: Set up Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.9
+      - name: Install docs requirements
+        run: |
+          pip install -r docs/source/requirements.txt
+
+      # Why do we have to do this?
```
Comment (mlodic) [inline .github/workflows/sphinx_build_on_release.yml:20]: why is this comment here?

**Review by mlodic — CHANGES_REQUESTED** [review]
hey, thanks for the PR!

This action should work not on release trigger because it would be too late.

Consider that we make releases each time we open PRs from `develop` to `master`, this action would need to be executed only for the PRs that we do from `develop` to `master`. In this way we can fix the docs if something is wrong there, before creating a new release.

Then, I am not sure that this works as intended.

In RTD, after having built the requirements as you did here, it executed the build with this command:

``` 
/home/docs/checkouts/readthedocs.org/user_builds/intelowl/envs/develop/bin/python -m sphinx -T -E -b html -d _build/doctrees -D language=en . _build/html
```

Can you please make a test in your repository by triggering this action and then give us proof that this would work?

Thank you

**Comment by devmrfitz** [comment]
@mlodic By a test, do you mean that I should re-introduce the error fixed in last release to verify if the test catches it or not?

File: .github/workflows/sphinx_build_on_release.yml (hunk 1 above)
Comment (devmrfitz) [reply .github/workflows/sphinx_build_on_release.yml:20]: Shouldn't the additional package also be specified in docs/requirements.txt ?

Comment (eshaan7) [reply .github/workflows/sphinx_build_on_release.yml:20]: yes it should be in the requirements.txt file

**Comment by devmrfitz** [comment]
Here's a link for the same: https://github.com/devmrfitz/IntelOwl/runs/4763166223?check_suite_focus=true

File: .github/workflows/sphinx_build_on_release.yml (hunk 1 above)
Comment (mlodic) [reply .github/workflows/sphinx_build_on_release.yml]: RTD already performs some installation by default:

```
 /home/docs/checkouts/readthedocs.org/user_builds/intelowl/envs/develop/bin/python -m pip install --upgrade --no-cache-dir mock==1.0.1 pillow==5.4.1 alabaster>=0.7,<0.8,!=0.7.5 commonmark==0.8.1 recommonmark==0.5.0 sphinx<2 sphinx-rtd-theme<0.5 readthedocs-sphinx-ext<2.2 
```

It does that before installing the requirements in the `docs` folder. So yeah, there works but in the github action is needed.

**Comment by mlodic** [comment]
> @mlodic By a test, do you mean that I should re-introduce the error fixed in last release to verify if the test catches it or not?

No. Considering that this is something that works only on the condition that you specify in the Action, you need to have that action triggered so we can test if it works or not. 

Either you do it in your repo and show us the results or you momentarily set that this action should trigger on PR against develop so we can see if it works here

**Comment by mlodic** [comment]
> Here's a link for the same: https://github.com/devmrfitz/IntelOwl/runs/4763166223?check_suite_focus=true

exactly, something like that but with the action working

**Comment by devmrfitz** [comment]
> > Here's a link for the same: https://github.com/devmrfitz/IntelOwl/runs/4763166223?check_suite_focus=true
> 
> exactly, something like that but with the action working

Does this look good https://github.com/devmrfitz/IntelOwl/actions/runs/1677927980 ?

**Comment by devmrfitz** [comment]
@mlodic Is it alright if the action runs on every PR to master?

File: .github/workflows/sphinx_build_on_release.yml (hunk 1 above)
Comment (devmrfitz) [reply .github/workflows/sphinx_build_on_release.yml]: @mlodic Is it alright to add it to docs/source/requirements.txt?

**Pull request merged**